c++/src/kj/async-win32.c++

```diff
@@ -158,13 +158,21 @@ Own<Win32EventPort::SignalObserver> Win32IocpEventPort::observeSignalState(HANDL
 }
 
 bool Win32IocpEventPort::wait() {
-  waitIocp(timerImpl.timeoutToNextEvent(clock.now(), MILLISECONDS, INFINITE - 1)
-      .map([](uint64_t t) -> DWORD { return t; })
-      .orDefault(INFINITE));
-
-  timerImpl.advanceTo(clock.now());
-
-  return receivedWake();
+  const bool woke = receivedWake();
+  if(woke) {
+    waitIocp(0);
+    return true;
+  }
+  try{
+    waitIocp(timerImpl.timeoutToNextEvent(clock.now(), MILLISECONDS, INFINITE - 1)
+        .map([](uint64_t t) -> DWORD { return t; })
+        .orDefault(INFINITE));
+
+    timerImpl.advanceTo(clock.now());
+  }catch(...) {
+    
+  }
+  return woke;
```

Comment:
`woke` is always false here... so this means we're now ignoring cross-thread wakes that occurred during the wait that we just completed. It would still appear to work since the _next_ wait would see the wakeup, but we shouldn't delay the notification.

```suggestion
  return receivedWake();
```